packages/wdio-local-runner/src/processFactory.ts

```diff
@@ -0,0 +1,53 @@
+import { spawn, fork } from 'node:child_process'
+import type { ChildProcess, SpawnOptions, ForkOptions } from 'node:child_process'
+import type { XvfbManager } from '@wdio/xvfb'
+import logger from '@wdio/logger'
+
+const log = logger('@wdio/local-runner:ProcessFactory')
+
+export interface ProcessCreator {
+    createWorkerProcess(scriptPath: string, args: string[], options: ProcessCreationOptions): ChildProcess
+}
+
+export interface ProcessCreationOptions {
+    cwd?: string
+    env?: Record<string, string>
+    execArgv?: string[]
+    stdio?: ('inherit' | 'pipe' | 'ignore' | 'ipc')[]
+}
+
+export class ProcessFactory implements ProcessCreator {
+    constructor(private xvfbManager: XvfbManager) {}
+
+    createWorkerProcess(scriptPath: string, args: string[], options: ProcessCreationOptions): ChildProcess {
+        const { cwd, env, execArgv = [], stdio } = options
+
+        if (this.xvfbManager.shouldRun()) {
+            log.debug('Creating worker process with xvfb-run wrapper')
+
+            // Use spawn with xvfb-run wrapper
+            const nodeArgs = [...execArgv, scriptPath, ...args]
+
+            return spawn('xvfb-run', [
+                '--auto-servernum',
+                '--',
+                'node',
+                ...nodeArgs
+            ], {
+                cwd,
+                env,
+                stdio
+            } as SpawnOptions)
+        }
+        log.debug('Creating worker process with regular fork')
```

Comment:
Is this something we can include the xvfb package?

Reply:
We could, but I thought it better to keep the process management in the local-runner.  Not too bothered though, it could certainly live in the xvfb package.

Reply:
Yeah, I think it makes conceptional more sense